ggshield/config.py

```diff
@@ -36,287 +28,266 @@ def replace_in_keys(data: Dict, old_char: str, new_char: str) -> None:
 
-    def __init__(self) -> None:
-        self.purge()
-        self.load_cache()
 
-    def __getattr__(self, name: str) -> Any:
-        # Required for dynamic types on mypy
-        return object.__getattribute__(self, name)
-
-    def get_attributes_keys(self) -> List:
-        return list(
-            list(zip(*self.attributes))[0]
-        )  # get list of first elements in tuple
-
-    def __setattr__(self, name: str, value: Any) -> None:
-        if isinstance(value, list):
-            attribute = getattr(self, name)
-
-            if isinstance(attribute, list):
-                for elem in value:
-                    if elem not in attribute:
-                        attribute.append(elem)
-            else:
-                getattr(self, name).update(value)
-        else:
-            super().__setattr__(name, value)
+@dataclass
+class HostConfig:
+    name: str
+    account: AccountConfig  # Only handle 1 account per host for the time being
+    default_token_lifetime: Optional[timedelta] = None
 
-    def load_cache(self) -> bool:
-        if not os.path.isfile(self.CACHE_FILENAME):
-            return True
+    @classmethod
+    def load(cls, data: Dict) -> "HostConfig":
+        assert len(data["accounts"]) <= 1
+        data["account"] = AccountConfig(**data["account"])
+        return cls(**data)
 
-        _cache: dict = {}
-        if os.stat(self.CACHE_FILENAME).st_size != 0:
-            try:
-                f = open(self.CACHE_FILENAME, "r")
-            except PermissionError:
-                # Hotfix: for the time being we skip cache handling if permission denied
-                return True
-            else:
-                with f:
-                    try:
-                        _cache = json.load(f)
-                        # Convert back all sets that were serialized as lists
-                        for attr in self.attributes:
-                            if type(attr.default) is set and attr.name in _cache:
-                                _cache[attr.name] = set(_cache[attr.name]) or set()
-                    except Exception as e:
-                        raise click.ClickException(
-                            "Parsing error while"
-                            f"reading {self.CACHE_FILENAME}:\n{str(e)}"
-                        )
-        self.update_cache(**_cache)
-        return True
 
-    def update_cache(self, **kwargs: Any) -> None:
-        for key, item in kwargs.items():
-            if key in self.get_attributes_keys():
-                if isinstance(item, list):
-                    attr = getattr(self, key)
-                    for elem in item:
-                        if elem not in attr:
-                            attr.append(elem)
-                else:
-                    setattr(self, key, item)
+def get_auth_config_dir() -> str:
+    dir_path: str = user_config_dir(appname="ggshield", appauthor="GitGuardian")
+    return dir_path
```

Comment:
Can be simplified to:
```suggestion
    return user_config_dir(appname="ggshield", appauthor="GitGuardian")
```

Reply:
no because mypy fails, is there a better way to do it? (mypy just feels like a waste of time).

Reply:
mypy is annoying sometimes, but it regularly catches mistakes, I find it quite useful.

In this case that's probably because the `appdirs` package does not ship with type hints. Someone created a package for this though: https://pypi.org/project/types-appdirs/, I think adding this to our packages (and to mypy config in the .pre-commit-config.yaml) should help.